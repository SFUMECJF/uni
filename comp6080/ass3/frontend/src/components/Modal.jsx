/**
 * Holds all Modal and things associated with modal
 */
import React, { useState } from 'react';
import { Modal, ModalBody, ModalFooter, ModalTitle, Button } from 'react-bootstrap';
import PropTypes from 'prop-types';

/**
 * Given a title string and content element,
 * will show a modal to the user that overlaps everything
 * @param {Objectr} props requires title, content element and bool show
 * @returns JSXElement modal
 */
export function CustomModal (props) {
  const [show, setShow] = useState(props.show);
  const title = props.title;
  const content = props.content;

  // handle close function toggle
  const handleClose = () => setShow(false);

  return (
    <>
      <Modal show={show} onHide={handleClose} backdrop='static'>
        <Modal.Header closeButton>
          <ModalTitle>{title}</ModalTitle>
        </Modal.Header>
        <ModalBody>{content}</ModalBody>
        <ModalFooter>
          <Button id='modalCloseButton' className='btn-secondary' onClick={handleClose}>
            Close
          </Button>
        </ModalFooter>
      </Modal>
    </>
  );
}

CustomModal.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.element.isRequired,
  show: PropTypes.bool.isRequired,
}
