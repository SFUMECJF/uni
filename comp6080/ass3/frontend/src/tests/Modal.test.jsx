/**
 * Tests modal!
 */
import { shallow } from 'enzyme';
import React from 'React';
import { CustomModal } from '../components/Modal.jsx';

// will create a modal with some tests stuff in it
// modal should have a title, have the content, have two methods of closing it
// and should not allow keyboard dismissal
// if modal can have content, it can have anything in it
// them most relied upon and reused component!!!
describe('Modal', () => {
  // test can mount
  // initial snapshot render check
  it('Modal shallow renders onto page', () => {
    const testRender = shallow(<CustomModal show={true} title={'Test'} content={<text>This is a test</text>}/>);
    expect(testRender).toMatchSnapshot();
  });

  const modal = shallow(<CustomModal show={true} title={'Test'} content={<text>This is a test</text>}/>);
  // test title
  it('modal title is "Test"', () => {
    const title = modal.find('ModalTitle');
    expect(title.text()).toEqual('Test');
  });
  it('check snapshot after title', () => {
    const title = modal.find('ModalTitle');
    expect(title).toMatchSnapshot();
  });

  // test contents
  it('Modal content is "This is a test"', () => {
    const content = modal.find('ModalBody');
    expect(content.text()).toEqual('This is a test');
  });
  it('Check snapshot after content', () => {
    const content = modal.find('ModalBody');
    expect(content).toMatchSnapshot();
  });

  // test close button
  it('Can close via pressable close button', () => {
    const closeButton = modal.find('Button');
    closeButton.simulate('click');
    // ensure that nothing has changed after clicking it
    expect(closeButton).toMatchSnapshot();
  })

  // test close x button at top
  it('Can close via x button at top', () => {
    const closeX = modal.find({ closeButton: true });
    closeX.simulate('click');
    expect(closeX).toMatchSnapshot();
  })
})
