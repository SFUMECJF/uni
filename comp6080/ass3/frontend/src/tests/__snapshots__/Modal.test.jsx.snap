// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Can close via pressable close button 1`] = `ShallowWrapper {}`;

exports[`Modal Can close via x button at top 1`] = `ShallowWrapper {}`;

exports[`Modal Check snapshot after content 1`] = `ShallowWrapper {}`;

exports[`Modal Modal shallow renders onto page 1`] = `ShallowWrapper {}`;

exports[`Modal check snapshot after title 1`] = `ShallowWrapper {}`;
